### DISCUSSION TITLE
Create mod deployer FakePlayer can't be trusted. Only with /trust public

### DISCUSSION THREAD

**Discussion by muriplz** [GriefPrevention/GriefPrevention#1305] [Ideas]
### What happened:

Deployers from Create mod can't place saplings inside claims. (when they are inside of course).
I tried /trust public, and is the only thing that works, i cant /trust the deployer's fake player with UUID 9e2faded-cafe-4ec2-c314-dad129ae971d.
### What was expected:
I expect the deployer to be able to place saplings, if not a main feature of the mod can be exposed.

### Steps to reproduce:
1. Install the Create mod 0.3 and grief prevention.
2. Make a claimed area.
3. Try to make a deployer place saplings inside it. (Easiest way: with Hand Crank)

### Server and GriefPrevention version:

Paste of `/version`: This server is running Mohist version 1.16.5-384 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT, Forge version 36.0.46)
Paste of `/version GriefPrevention`:  GriefPrevention v16.17.1-2b69191

### Stack trace/error or server log

There is no logs, it just won't place it.

### Debugging steps attempted

- [ ] I attempted running only GriefPrevention on the server
  - [ ] Issue still occurs even if GriefPrevention is the only plugin running
- [ ] I attempted testing for the issue on a new server
  - [ ] Issue still occurs even after testing on a new server
- [ ] I attempted testing for the issue with the vanilla client (no mods)
  - [ ] Issue still occurs even after testing with the vanilla Minecraft client

**Comment by RoboMWM** [comment]
Mod configuration was dropped with the MC 1.13 material flattening. You'll need an addon to deal with these.

**Comment by Eriador26** [comment]
Is there a solution already to this?

**Comment by Noxfield** [comment]
I'm having the same problem, I don't know why Eriador26 was marked as off topic for asking if a solution was found/known. I'm not very aware of the "material flattening" either, nor how that affects the way grief prevention recognizes players. More insight would be nice.

**Reply by Jikoo** [reply]
GP uses Bukkit's system for obtaining players via lookup by name as necessary. Since the server implementation does not provide the dummy players used by mods as valid past players for Bukkit plugins, GP does not have the capacity to look them up.

The material flattening is when Mojang moved to using keys for materials instead of numeric IDs. GP stopped offering explicit mod support in general at that time - if a server wants to implement the Bukkit API with another modding platform, it's on the server implementation to play nice with standard Bukkit plugin practice.

It is possible for an addon to trust mods by directly trusting UUIDs (which would probably be done via an alias mapping), but to my knowledge no such addon exists at this time.

I wasn't the one to mark Eriador26 as offtopic, however, I suspect it was because they asked here after creating a separate discussion on the same topic and receiving the same response as Robo's reply here. Spam goes in the spam bin.

**Reply by Noxfield** [reply]
Cool, thank you. And thank you for the quick, very late night reply.

**Comment by Jikoo** [comment]
Years later, so I'm assuming you either elected not to use GP or have rolled your own trust solution, but this has finally been resolved to some degree. With #2045 you can now use UUIDs with `/trust`. An addon would still do a better job of this (i.e. have a mapping of mod names and fake player UUIDs so players would not have to be aware of the fake player UUIDs) but it is now at least possible to trust Create out of the box.
If you're looking to make this more user-friendly you can use the Bukkit commands.yml feature (assuming your hybrid server implementation supports it) to add commands to manage particular mods.

For example, the UUID of Create in OP's modpack is `9e2faded-cafe-4ec2-c314-dad129ae971d`. Communicating `/trust 9e2faded-cafe-4ec2-c314-dad129ae971d` and `/untrust 9e2faded-cafe-4ec2-c314-dad129ae971d` to users is hard, but telling them to `/trustcreate` or `/untrustcreate` is easy to remember.
```yaml
aliases:
  trustcreate:
    - "trust 9e2faded-cafe-4ec2-c314-dad129ae971d"
  untrustcreate:
    - "untrust 9e2faded-cafe-4ec2-c314-dad129ae971d"
```
